src/System.Collections/src/System/Collections/Generic/HashSet.cs

```diff
@@ -265,6 +266,13 @@ public bool Contains(T item)
                     {
                         return true;
                     }
+
+                    if (collisionCount >= _slots.Length)
+                    {
+                        // The chain of entries forms a loop; which means a concurrent update has happened.
+                        throw new InvalidOperationException(SR.InvalidOperation_ConcurrentOperationsNotSupported);
```

Comment:
We should probably use a ThrowHelper scheme instead of throw for these as some of these loops are particularly hot paths. There is not a ThrowHelper.cs in CoreFX common (probably should be). You can copy one of the several sprinkled around CoreFX sources.

Reply:
> There is not a ThrowHelper.cs in CoreFX common (probably should be)

https://github.com/dotnet/corefx/pull/26636#discussion_r165062616